lib/jobs/emc-redfish-catalog.js

```diff
@@ -90,30 +91,81 @@ function EmcRedfishCatalogJobFactory(
         })
         .then(function(id) {
             assert.string(id, 'Element Identifier');
-            return redfish.clientRequest(id);
+            return self.redfish.clientRequest(id);
         })
         .then(function(res) {
             assert.object(res, 'Element Resource Object');
             var elements = _.get(res, 'body.Members');
             return elements;
         })
         .map(function(element) {
-            return redfish.clientRequest(element['@odata.id']);
+            return self.redfish.clientRequest(element['@odata.id']);
         })
         .map(function(element) {
             assert.object(element, 'Element Resource Object');
             assert.object(element.body);
             return element.body;
         })
-        .then(function(data) {
-            assert.ok(Array.isArray(data), 'Element Data');
+        .map(function(element) {
+            return self.redfish.clientRequest(element['@odata.id'])
+                .then (function(element){
+                    return element;
+                });
+        })
+        .map(function(element) {
+            assert.object(element, 'Element Resource Object');
+            assert.object(element.body);
+            return element.body;
+        })
+        .map(function(element){
+            _.forEach(element,function(value,key){
+                if(key === "Processors" || key === "Vpd" || key === "Dimms" || key === "Drives") {
+                    element = Promise.resolve(element)
+                        .then(function(element){
+                           return  self.updatingComponents(element,key);
+                        });
```

Comment:
The `Promise.resolve(element)` is abundant here, let `element = self.updatingComponents(element,key)` is OK.